### PR TITLE
VB-4107 - Add advanceFromDateByDays as a request parameter to available visit sessions.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt [hunk 1]
```diff
@@ -130,8 +130,20 @@ class OrchestrationSessionsController(private val visitSchedulerSessionsService:
...
   ): List<AvailableVisitSessionDto> =
-    visitSchedulerSessionsService.getAvailableVisitSessions(prisonCode, prisonerId, sessionRestriction, visitors, withAppointmentsCheck ?: true, excludedApplicationReference)
+    visitSchedulerSessionsService.getAvailableVisitSessions(
+      prisonCode = prisonCode,
+      prisonerId = prisonerId,
+      requestedSessionRestriction = sessionRestriction,
+      visitors = visitors,
+      withAppointmentsCheck = withAppointmentsCheck ?: true,
+      excludedApplicationReference = excludedApplicationReference,
+      advanceFromDateByDays = advanceFromDateByDays ?: 0,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt:145]: no need if you have above

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonService.kt [hunk 2]
```diff
@@ -54,21 +56,6 @@ class PrisonService(
     prisonCode: String,
   ): DateRange {
     val prison = visitSchedulerClient.getPrison(prisonCode)
-    return getToDaysDateRange(prison)
-  }
-
-  fun getToDaysDateRange(
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonService.kt:60]: nice

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitSchedulerSessionsService.kt [hunk 3]
```diff
@@ -41,10 +43,15 @@ class VisitSchedulerSessionsService(
     visitors: List<Long>?,
     withAppointmentsCheck: Boolean,
     excludedApplicationReference: String? = null,
+    advanceFromDateByDays: Int = 0,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitSchedulerSessionsService.kt:46]: I think this should be in one place for all, maybe leave it out for others ?

**Review by AledWEvans — COMMENTED** [review]
some comments but most all good

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationSessionsController.kt:145]: I had to keep this but remove it from above.

**Review by AledWEvans — DISMISSED** [review]
LGTM

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**